source/RenderBuffer.h

```diff
@@ -0,0 +1,92 @@
+/* RenderBuffer.h
+Copyright (c) 2023 by thewierdnut
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef RENDERBUFFER_H
+#define RENDERBUFFER_H
+
+#include "Point.h"
+#include "Screen.h"
+
+
+
+// Class that can redirect all drawing commands to an internal texture.
+// This buffer uses coordinates from (0, 0) in the top left, to (width, height)
+// in the bottom right.
+class RenderBuffer
+{
+public:
+	// RAII wrapper to prevent accidentally not unbinding the render target.
+	class RenderTargetGuard final
+	{
+	public:
+		~RenderTargetGuard();
+
+		// Explicitly deactivate render target;
+		void Deactivate();
+
+
+	protected:
+		RenderTargetGuard(RenderBuffer &b, int screenWidth, int screenHeight);
+
+
+	private:
+		RenderBuffer &buffer;
+		Screen::ScreenDimensionsGuard screenGuard;
+		friend class RenderBuffer;
+	};
+
```

Comment:
```suggestion

public:
```

Reply:
Why do you want the public keyword a second time?